xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm

```diff
@@ -117,17 +117,19 @@
         #set($hasTagsPlugin = false)
         #set($tags = $doc.getTagList())
     #end
-    <div class="doc-tags #if(!$hasedit && $tags.isEmpty() )hidden#end" id="${tagsId}">
+    <div class="doc-tags#if(!$hasedit && $tags.isEmpty()) hidden#end" id="${tagsId}">
         $services.localization.render('core.tags.list.label')
         #if(!$tags.isEmpty())
             #foreach($tag in $tags)
                 #displayTag($tag)
             #end
         #end
-        #if($hasedit)
-            #if($xwiki.tag)
-                <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
-            #end
+        #if($hasedit && $xwiki.tag)
+            <div class="tag-tool tag-add">
+                #if("$!{request.showTagAddForm}" == '')
+                    <a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>
+                #{else}#displayAddForm()#end
```

Comment:
Note that this will probably generate some useless extra new lines in the generated HTML. You should keep it on one line (as suggested by Marius previously). What I suggested was to introduce some velocity variables. Thanks

Reply:
@vmassol Can you tell me what variables I need to add? I didn't quite understand what Marius said above about changing just one line of code

Reply:
The goal is to make the code the most readable possible. Always a good thing to do :)

For example, something like:

```
#set ($url = "$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")")
#if("$!{request.showTagAddForm}" == '')
  <a href="$url" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>
#else
  #displayAddForm()
#end
```

BTW I'm not sure about the extra new lines since the velocity `#if` shouldn't add a new line I think, did you check if that was the case or not?

Reply:
Ok, I'll work on this! Thanks for the hint

Reply:
@CatalinStratu my comment was not about the line that generates the + link. My comment was about **all your changes** done to the ``documentTags.vm``: there are **too many** and I'm not convinced that all of them are needed to fix the problem. I'll repeat: all you need to do is to wrap the line (a single line!) that outputs "Tags:" with a ``#if`` so that it gets displayed only when the user has edit right (can add tags) or there are existing tags. So that means one line removed and 3 lines added. I really don't see how I can be more explicit than than, except by making the change myself..

Reply:
@mflorea 
` #if(!$hasedit && $tags.isEmpty())
         $services.localization.render('core.tags.list.label')
    #end`